### DISCUSSION TITLE
Static abstract init for interfaces

### DISCUSSION THREAD

**Discussion by budgetdevv** [dotnet/csharplang#8470] [Language Ideas]
```cs
        public interface IFoo
        {
            public static abstract bool IsTrue { init; }
        }

        public struct Foo: IFoo
        {
            public static readonly bool IsTrue = true;
        }
```

Basically a way to define static readonly as a contract. It can be useful for performance optimization, as static readonly is treated as a runtime constant I believe.

**Comment by siegfriedpammer** [comment]
How should the compiler enforce the initialization of a static property?

Static constructors are executed when the runtime loads the type info for the first time, but that can happen transparently because there are no user-supplied arguments.

Where would you expect an error to be reported, if you forgot to initialize the type and *the only* use is in a typeof-expression in an attribute?
